auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -201,7 +205,11 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     private void finish(int resultCode) {
-        mActivity.finish(RESULT_OK, mActivity.getIntent());
+        mActivity.finish(resultCode, mActivity.getIntent());
+    }
+
+    private void finish(int resultCode, Intent data) {
+        mActivity.finish(resultCode, data);
```

Comment:
I see, but since the resultCode was being ignored `finish(RESULT_CANCELLED)` like on line 167 would result in `finish(RESULT_OK)` being called.  Maybe it would make sense to change line 167 and line 174 to call `finish(RESULT_OK)` directly instead?